File: android/app/src/main/kotlin/com/sangamshrestha/jett/EventListener.kt
package com.sangamshrestha.jett

import FilesStreamHandler
import PigeonEventSink
import PlatformFile

class EventListener : FilesStreamHandler() {
    private var eventSink: PigeonEventSink<List<PlatformFile>>? = null

    override fun onListen(p0: Any?, sink: PigeonEventSink<List<PlatformFile>>) {
        eventSink = sink
    }

    fun onEvent(files: List<PlatformFile>) = eventSink?.success(files)

    override fun onCancel(p0: Any?) {
        eventSink?.endOfStream()
        eventSink = null
    }
}

File: ios/Runner/EventListener.swift
class EventListener: FilesStreamHandler {
    var eventSink: PigeonEventSink<[PlatformFile]>?

    override func onListen(withArguments arguments: Any?, sink: PigeonEventSink<[PlatformFile]>) {
        eventSink = sink
    }

    func onEvent(files: [PlatformFile]) {
        eventSink?.success(files)
    }

    override func onCancel(withArguments arguments: Any?) {
        eventSink?.endOfStream()
        eventSink = nil
    }
}
